File: Yo/cursos/udemy/udemyCourseTranscriptionRecovery/playbooks/analyze_course.js
const fs = require('fs');
const path = require('path');

const outputDir = 'c:\\git\\wb_pr\\projects\\Yo\\cursos\\udemy\\udemyCourseTranscriptionRecovery\\playbooks\\output';
const files = fs.readdirSync(outputDir).filter(f => f.endsWith('.txt') && f.includes(' - '));

// Sort by numeric prefix
files.sort((a, b) => {
    const idA = parseInt(a.split(' - ')[0]);
    const idB = parseInt(b.split(' - ')[0]);
    return idA - idB;
});

const report = [];

files.forEach(filename => {
    const filePath = path.join(outputDir, filename);
    const lectureId = filename.split(' - ')[0];

    const content = fs.readFileSync(filePath, 'utf-8');
    const lines = content.split('\n').map(l => l.trim()).filter(l => l.length > 0);
    const headerSnippet = lines.slice(0, 10).join(' ');

    // Try to find "módulo de X"
    const moduleMatch = headerSnippet.match(/módulo de ([^,.\n]+)/i);
    const module = moduleMatch ? moduleMatch[1].trim() : "General";

    // Try to find a title-like sentence
    const titleMatch = headerSnippet.match(/vamos a (?:ver|hablar|aprender|empezar|explicar) (?:sobre |con |de |)(?:el |la |)([^,.\n]+)/i);
    let title = titleMatch ? titleMatch[1].trim() : "Untertitled Lecture";

    // Fallback for title
    if (title.length < 3 || /este vídeo|este nuevo vídeo|esta parte/i.test(title)) {
        title = lines[1] && lines[1].length > 5 ? lines[1] : (lines[0] || "Lecture " + lectureId);
    }

    report.push({
        id: lectureId,
        module: module,
        title: title
    });
});

console.log(`Total files: ${report.length}`);
report.forEach(item => {
    console.log(`${item.id} | ${item.module.padEnd(20).substring(0, 20)} | ${item.title.padEnd(40).substring(0, 40)}`);
});

// Write to a temporary JSON for later use
fs.writeFileSync('c:\\git\\wb_pr\\projects\\Yo\\cursos\\udemy\\udemyCourseTranscriptionRecovery\\playbooks\\course_metadata.json', JSON.stringify(report, null, 2));

File: Yo/cursos/udemy/udemyCourseTranscriptionRecovery/playbooks/analyze_course.py
import os
import re

output_dir = r"c:\git\wb_pr\projects\Yo\cursos\udemy\udemyCourseTranscriptionRecovery\playbooks\output"
files = [f for f in os.listdir(output_dir) if f.endswith(".txt") and " - " in f]

# Sort by numeric prefix
files.sort(key=lambda x: int(x.split(" - ")[0]) if x.split(" - ")[0].isdigit() else 0)

report = []

for filename in files:
    path = os.path.join(output_dir, filename)
    lecture_id = filename.split(" - ")[0]
    
    with open(path, "r", encoding="utf-8") as f:
        # Read first 5 lines
        lines = [f.readline().strip() for _ in range(5)]
        content = " ".join(lines)
        
        # Try to find "módulo de X"
        module_match = re.search(r"módulo de ([^,.\n]+)", content, re.IGNORECASE)
        module = module_match.group(1).strip() if module_match else "General"
        
        # Try to find a title-like sentence
        # Look for "vamos a ver X", "vamos a hablar de X", etc.
        title_match = re.search(r"vamos a (?:ver|hablar|aprender|empezar|explicar) (?:sobre |con |de |)(?:el |la |)([^,.\n]+)", content, re.IGNORECASE)
        title = title_match.group(1).strip() if title_match else "Untertitled Lecture"
        
        # Fallback for title if it's too generic
        if len(title) < 3 or title.lower() in ["este vídeo", "este nuevo vídeo", "esta parte"]:
             title = lines[1] if len(lines) > 1 and len(lines[1]) > 5 else lines[0]

        report.append({
            "id": lecture_id,
            "filename": filename,
            "module": module,
            "title": title,
            "first_line": lines[0],
            "second_line": lines[1]
        })

# Print a summary
print(f"Total files: {len(report)}")
for item in report:
    print(f"{item['id']} | {item['module'][:20]:20} | {item['title'][:40]:40}")
